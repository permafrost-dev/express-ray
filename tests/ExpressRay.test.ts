import { FakeExpressRay } from '@tests/TestClasses/FakeExpressRay';

it('sends requests to Ray', () => {
    const rayInstance = FakeExpressRay.create();

    const request = {
        method: 'GET',
        url: '/',
        params: {},
        query: {},
        headers: {},
    };

    rayInstance.request(request);

    expect(rayInstance.sentPayloads).toHaveLength(1);
    expect(rayInstance.sentPayloads).toMatchSnapshot();
});
