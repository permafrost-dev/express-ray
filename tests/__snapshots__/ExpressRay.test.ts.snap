// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sends requests to Ray 1`] = `
Array [
  Object {
    "meta": undefined,
    "payloads": TablePayload {
      "data": Object {
        "content": "",
        "origin": Object {
          "file": "",
          "function_name": "",
          "hostname": "remote",
          "line_number": 0,
        },
        "type": "",
      },
      "initialized": false,
      "label": "Request",
      "localPath": null,
      "remotePath": null,
      "values": Object {
        "headers": Object {},
        "method": "GET",
        "params": Object {},
        "query": Object {},
        "url": "/",
      },
    },
  },
]
`;
